xsdata/formats/dataclass/serializers/xml.py

```diff
@@ -90,11 +90,38 @@ def render_node(self, obj, parent, namespaces: Namespaces) -> Element:
                 else:
                     self.render_sub_nodes(parent, value, var, namespaces)
             else:
-                self.set_nil_attribute(parent, var, namespaces)
+                self.set_nil_attribute(parent, var.is_nillable, namespaces)
 
         self.unset_nil_attribute(parent)
         return parent
 
+    def next_value(self, vars: List[ClassVar], obj: Any):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'vars'](https://app.codacy.com/manual/tefra/xsdata/pullRequest?prid=5244803)